### DISCUSSION TITLE
Rack Images not showing.

### DISCUSSION THREAD

**Discussion by ssigle** [netbox-community/netbox#10815] [Q&A] [answered]
I am having an issue with my rack images not showing in Netbox v3.3.6, Python v3.8.10.

I've tried to select Images, and Images and Labels and neither will display my racks. Is there a way to fix this? Or has anyone had a similar issue?

When I click Download SVG, the rack does show in a new window. So it seems there's just an issue with the page.

I've tried creating new racks and I am still having the same issue as well.

![Netbox1](https://user-images.githubusercontent.com/87151221/199500170-e3bdfeeb-23fd-4c81-8ac4-6c161bfdf438.png)

**Comment by candlerb** [comment]
Works for me. Which browser are you using?  Can you try opening your browser's developer console and then refreshing the page - do you see any errors in the browser console?

**Reply by candlerb** [reply]
And just to be clear, the page you're looking at is `/dcim/racks/NNN/` ?

Also: the rack image is one of those few places where Netbox web rendering calls back to its own API endpoints. Look at the page source and you should see something like this where the rack image goes:

```
<object data="/api/dcim/racks/1/elevation/?face=front&amp;render=svg" class="rack_elevation">...</object>
```

However, the button to "Download SVG" is the same:

```
<a class="btn btn-outline-primary btn-sm" href="/api/dcim/racks/1/elevation/?face=front&amp;render=svg">
        <i class="mdi mdi-file-download"></i> Download SVG
    </a>
```

**Comment by ssigle** [comment]
This is the only issue appearing in the console.

![Netbox2](https://user-images.githubusercontent.com/87151221/199725359-c767b034-1c60-4504-9a6a-f1e4ccf552fb.png)

Currently, I am looking at my /dcim/racks/2/ page, but now as I am looking through things.. It's also not showing the images for the cable traces as well.

Yes, I am seeing the <object data="/api/dcim/racks/2/elevation/?face=front&amp;render=svg" class="rack_elevation"></object> and <object data="/api/dcim/racks/2/elevation/?face=rear&amp;render=svg" class="rack_elevation"></object> Elements.

**Reply by candlerb** [reply]
Can you try the "console" tab rather than "issues"? And is there a "network" tab which shows external accesses?

Then you can see whether the `/api/dcim/racks/N/elevation/` endpoint is being accessed, and if it is, what the response code is.

**Comment by ssigle** [comment]
![Netbox4](https://user-images.githubusercontent.com/87151221/199729262-8a6a6dbb-8d60-4058-ad93-d280c6bc3681.png)
![Netbox5](https://user-images.githubusercontent.com/87151221/199729265-10a564de-a358-4eda-803d-27457a7118aa.png)

Sorry, I don't exactly know where to go to tell if it is being accessed.. but I can browse to that and it gives me a HTTP 200 OK.

**Reply by kkthxbye-code** [reply]
Your browser or an extension is blocking the svg embed. It's not a netbox issue.

**Reply by ssigle** [reply]
Its doing it in Microsoft Edge also.

**Reply by kkthxbye-code** [reply]
I can only help you with the information you provide. If you think this is a netbox issue (it's not) and it's present in edge also, you need to show the console and the network tab for edge then. In your image there it clearly says "blocked: other" which is what devtools displays if a request is blocked by an adblocker for example.

I just downloaded brave and tested and it works fine. This is a problem with your netbox instance or your browser and if you want help you need to provide more information.

Did you try from another machine? Did you try from chrome? Did you try from incognito mode? Did you try the demo instance https://demo.netbox.dev/ ?

**Reply by ssigle** [reply]
The demo site shows the racks just fine in Brave on my end.
![Netbox7](https://user-images.githubusercontent.com/87151221/199738276-251bd525-93d1-4a55-98df-b1cde43949a0.png)

Here are the tabs of the Console and the Network of Edge browser.
![Netbox8](https://user-images.githubusercontent.com/87151221/199740177-e506e235-d30c-4cd5-b9c3-fd2859f3cc8e.png)
![Netbox9](https://user-images.githubusercontent.com/87151221/199740179-a8453149-2157-49a1-b9c9-fa3e71ed6a3a.png)

Here are the incognito modes of Brave and Edge
![Netbox10](https://user-images.githubusercontent.com/87151221/199740299-3f1e6e2b-a66b-4456-aab2-eb708cd67456.png)
![Netbox11](https://user-images.githubusercontent.com/87151221/199740302-26544849-8499-4d55-8f51-8684a9870d8c.png)

Screenshot from another machine.
![Netbox12](https://user-images.githubusercontent.com/87151221/199740988-5c177e23-7480-487d-82f5-7f73151f5d1c.png)

Shields are off on my brave browser and the only extension installed is LastPass.

I just installed Chrome, attached are screen shots in a normal browsing session and an incognito one.
![Netbox13](https://user-images.githubusercontent.com/87151221/199742165-b60ccc3e-f6d2-49dc-9fec-113407c22eac.png)
![Netbox14](https://user-images.githubusercontent.com/87151221/199742166-74bc235d-f09a-4a81-a1de-bd301565175b.png)

**Reply by kkthxbye-code** [reply]
In the network tab of one of the browsers press one of the svg requests ang go to the response tab. See what is there. The request is returning ~500 bytes of data which is way too little. Best guess is improperly configured webserver, but again there's barely any information about your setup.

**Reply by ssigle** [reply]
<svg xmlns='http://www.w3.org/2000/svg' width='8' height='8'><path d='M2.5 0L1 1.5 3.5 4 1 6.5 2.5 8l4-4-4-4z' fill='currentColor'/></svg>

I'll dig into my webserver settings and see if I am missing something.

**Reply by ssigle** [reply]
Issue is with my ssl config... You can close this. I just removed the line and I'll work on that later.


Thanks for the help!

**Reply by hennning** [reply]
@ssigle I have the same issue as you.
What did you do to solve the issue?

I'm running standard docker install, but I'm running it behind a traefik proxy if that makes any difference.

**Reply by candlerb** [reply]
As was already explained, you need to look in your browser's developer tools:

* in the console, look for Javascript or HTTP errors
* in the network tab, look to see what response you're getting to the `/api/dcim/racks/NNN/elevation/?face=front&amp;render=svg` request.

Then it should be possible to work out what's going on.

**Reply by ssigle** [reply]
> @ssigle I have the same issue as you. What did you do to solve the issue?
> 
> I'm running standard docker install, but I'm running it behind a traefik proxy if that makes any difference.

It ended up being how I setup my SSL config in the webserver. I commented it out for now until I have time to work on it.

**Comment by ssigle** [comment]
I forgot to add, I am using brave browser.

**Comment by ghost** [comment, marked as answer]
As @ssigle, I was having the same issue as before. After an extended amount of time, I was able to find a solution to this problem. This solution may help.

So, how did I find the solution?
After reading through the post, and following the @candlerb steps, I noticed that under the Element referring to the object type, I wasn't getting the expected **200 OK** status, but a **(blocked:other)** instead. 

Subsequently, I did exactly what @ssigle did to "temporarily solve" his problem, which was to disable SSL:443 under my nginx.conf file. Lo and behold, the problem got solved, so I knew that there was something going on with SSL.

My next step was to enable both HTTP and HTTPS, so that I could login to the netbox by using SSL and Plain-Text.

On one Tab within my Google Chrome, I accessed my netbox using: http://127.0.0.1/. Additionally, on a different Tab, I also accessed my netbox using https://127.0.0.1/ 

After successfully login into the website, I started comparing both Network Elements to see what was the Response header when attempting to do the API GET, to display the server rack image.

I found out about a header named: **X-Frame-Options** and here is what I saw:

-For the http connection:  **X-Frame-Options: SAMEORIGIN**
-For the https connection: **X-Frame-Options: DENY**

It was a curious finding, and after researching the X-Frame-Option and from where this header was coming from, then I traced back the trail back to the _ssl.config_ that I have created to be used alongside my nginx config file when using the Diffie-hellmen group. Inside the ssl_config file, I found this:
````
$ sudo vim ssl_config.conf

....
add_header X-Frame-Options DENY;
add_header X-Content-Type-Options nosniff;

ssl_dhparam /etc/ssl/certs/dhparam.pem
````

Bob's your uncle! I commented the line containing: "#add_header X-Frame-Options DENY", reload nginx and the problem finally got solved, the rack images are not being displayed using HTTPS.

**Reply by ghost** [reply]
thank you. i was having the exact same issue. I commented out the same line and started working for me too.
your post should be marked as the answer.
